nox/manifest.py

```diff
@@ -184,6 +184,19 @@ def make_session(
             func.should_warn[WARN_PYTHONS_IGNORED] = func.python
             func.python = False
 
+        if self._config.extra_pythons:
+            # if extra python is provided, expand the func.python list to
+            # include additional python interpreters
+            if isinstance(func.python, (list, tuple, set)):
+                func.python.extend(self._config.extra_pythons)
+            elif not multi:
+                # if this is multi, but there is only a single interpreter,
+                # it is the reentrant case.  The extra_python interpreter
+                # shouldn't be added in that case. Otherwise, add the extra
+                # specified python.
+                func.python = [func.python]
+                func.python.extend(self._config.extra_pythons)
+
```

Comment:
The suggestion below should take care of the case pointed out by @pradyunsg, where the extra Python is already present in `func.python`, using set union.

There is also another edge case where `func.python` is `False` or `None`, meaning that the session should run without a backend. My intuition would be that we should skip that case because the user intention is that no virtual environment should be created for the session. Does that make sense?

Everything else is just to make the type checker happy:

- Members of `argparse.Namespace` have the type `Any`, so we use a variable annotation to refine the type. Use a comment for this to avoid confusing Python 3.5.
- We assume that `func.python` is either a list, set, tuple, None, False, or a string. The type checker thinks it could also be True or some unrelated sequence type. The assertion rules that out. Not sure if we have checks in place for the case that users do use those other types in their Nox files. (They should get a nice error message. But this is probably the wrong place for that kind of validation.)
- I also had to wrap the set literal in `list`. Our types use `typing.Sequence` here, and sets are not sequences because they're not indexable. That type should probably be fixed at some point, as the code clearly assumes that sets are legal.

```suggestion
        if self._config.extra_pythons:
            # if extra python is provided, expand the func.python list to
            # include additional python interpreters
            extra_pythons = self._config.extra_pythons  # type: List[str]
            if isinstance(func.python, (list, tuple, set)):
                func.python = list({*func.python, *extra_pythons})
            elif not multi and func.python:
                # if this is multi, but there is only a single interpreter, it
                # is the reentrant case. The extra_python interpreter shouldn't
                # be added in that case. If func.python is False, the session
                # has no backend; treat None and the empty string equivalently.
                # Otherwise, add the extra specified python.
                assert isinstance(func.python, str)
                func.python = list({func.python, *extra_pythons})

```